### PR TITLE
fix: Scroll to input when focusing keyword filter

### PR DESCRIPTION
**Pull request by olerichter00**
The type of this PR is: **Enhancement**

This PR resolves [CX-1743]

### Description

https://user-images.githubusercontent.com/4691889/126983767-54027d74-ef7c-47cb-812e-9d8018d5b283.mp4

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Scroll to keyword filter input when keyboard opens on artist insights screen - ole

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[CX-1743]: https://artsyproduct.atlassian.net/browse/CX-1743

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 2e0293f54c90eda534c663dd68d5c96aa351321a
</p>

File: src/lib/Components/Artist/ArtistInsights/ArtistInsights.tsx [hunk 1]
```diff
@@ -79,17 +90,20 @@ export const ArtistInsights: React.FC<ArtistInsightsProps> = (props) => {
         innerRef={flatListRef}
       >
         <MarketStatsQueryRenderer artistInternalID={artist.internalID} environment={relay.environment} />
-        <View
+        <ScrollView
```
Comment (olerichter00) [inline src/lib/Components/Artist/ArtistInsights/ArtistInsights.tsx:93]: Not sure if we can/should use a `ScrollView` inside of another scroll view 🤔

File: src/lib/Components/Artist/ArtistInsights/ArtistInsights.tsx [hunk 2]
```diff
@@ -54,11 +57,19 @@ export const ArtistInsights: React.FC<ArtistInsightsProps> = (props) => {
...
-    flatListRef.current?.getNode().scrollToOffset({ animated: true, offset: auctionResultsYCoordinate.current })
-  }, [auctionResultsYCoordinate])
+    let auctionResultYOffset = auctionResultsYCoordinate.current
+
+    // if we scroll up less than SCROLL_UP_TO_SHOW_THRESHOLD the header won't expand and we need another offset
+    if (contentYScrollOffset.current - 2 * auctionResultYOffset <= SCROLL_UP_TO_SHOW_THRESHOLD) {
+      auctionResultYOffset += ARTIST_HEADER_HEIGHT
+    }
+    flatListRef.current?.getNode().scrollToOffset({ animated: true, offset: auctionResultYOffset })
+  }, [auctionResultsYCoordinate, contentYScrollOffset])
```
Comment (olerichter00) [inline src/lib/Components/Artist/ArtistInsights/ArtistInsights.tsx:67]: **Note:** This also fixes a bug where we didn't scroll up enough when scrolling doesn't force the header to be expanded (scrolling up less than SCROLL_UP_TO_SHOW_THRESHOLD).

**Review by pvinis — APPROVED** [review]
Awesome work!
I reverted one file because it causes CI to break with these changes. I would have asked but in order to merge soon, I did it.

🚢

**Pull request merged**